spatialos-sdk/src/worker/schema/bundle.rs

```diff
@@ -0,0 +1,620 @@
+            })
+        }
+    }
+
+    pub fn dump_component_update(
+        &self,
+        component_id: ComponentId,
+        src: &mut SchemaComponentUpdate,
+    ) -> JsonConversionResult<String> {
+        unsafe {
+            self.dump_generic(move || {
+                Schema_Json_DumpComponentUpdate(self.ptr.as_ptr(), component_id, src.as_ptr_mut())
+            })
+        }
+    }
+
+    pub fn load_command_request<T: AsRef<str>>(
+        &self,
+        component_id: ComponentId,
+        command_index: u32,
+        json: T,
+    ) -> JsonConversionResult<Owned<SchemaCommandRequest>> {
+        let json =
+            CString::new(json.as_ref()).map_err(|_| "Null byte found in 'json'".to_string())?;
+
+        unsafe {
+            self.load_generic(move || {
+                Schema_Json_LoadCommandRequest(
+                    self.ptr.as_ptr(),
+                    component_id,
+                    command_index,
+                    json.as_ptr(),
+                )
+            })
+        }
+    }
+
+    pub fn dump_command_request(
+        &self,
+        component_id: ComponentId,
+        command_index: u32,
+        src: &mut SchemaCommandRequest,
+    ) -> JsonConversionResult<String> {
+        unsafe {
+            self.dump_generic(move || {
+                Schema_Json_DumpCommandRequest(
+                    self.ptr.as_ptr(),
+                    component_id,
+                    command_index,
+                    src.as_ptr_mut(),
+                )
+            })
+        }
+    }
+
+    pub fn load_command_response<T: AsRef<str>>(
+        &self,
+        component_id: ComponentId,
+        command_index: u32,
+        json: T,
+    ) -> JsonConversionResult<Owned<SchemaCommandResponse>> {
+        let json =
+            CString::new(json.as_ref()).map_err(|_| "Null byte found in 'json'".to_string())?;
+
+        unsafe {
+            self.load_generic(move || {
+                Schema_Json_LoadCommandResponse(
+                    self.ptr.as_ptr(),
+                    component_id,
+                    command_index,
+                    json.as_ptr(),
+                )
+            })
+        }
+    }
+
+    pub fn dump_command_response(
+        &self,
+        component_id: ComponentId,
+        command_index: u32,
+        src: &mut SchemaCommandResponse,
+    ) -> JsonConversionResult<String> {
+        unsafe {
+            self.dump_generic(move || {
+                Schema_Json_DumpCommandResponse(
+                    self.ptr.as_ptr(),
+                    component_id,
+                    command_index,
+                    src.as_ptr_mut(),
+                )
+            })
+        }
+    }
+
+    unsafe fn load_generic<
+        T: OwnedPointer + ToOwned<Owned = Owned<T>>,
+        F: FnOnce() -> *mut T::Raw,
+    >(
+        &self,
+        load: F,
+    ) -> JsonConversionResult<Owned<T>> {
+        let data = load();
```

Comment:
What's the reason for `load` being a function? If it's a `FnOnce` that doesn't take an parameters and is called immediately within `load_generic`, wouldn't it be the same to have the code calling `load_generic` directly pass the result of the callback into `load_generic`, instead of passing in an `FnOnce`? The same goes for `dump_generic` below.

Reply:
Yeah this seems like some overzealous refactoring. Will simplify as suggested :)